src/main/java/core/basesyntax/service/FruitShopService.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.UnknownOperationException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopService {
+    private Map<String, Integer> fruitInventory = new HashMap<>();
+    private Map<OperationType, OperationHandler> operationStrategy = new HashMap<>();
+
+    public FruitShopService(Map<OperationType, OperationHandler> operationStrategy) {
+        this.operationStrategy = operationStrategy;
```

Comment:
There is no need to reassign the 'operationStrategy' map in the constructor if you are passing it as a parameter. You can remove the initialization on line 14 and make the field final to ensure it is only set once.